File: 151004/Bordilovsky/DC_NoSQL/Publisher/DTOs/Request/UserRequestTo.cs
namespace DC_REST.DTOs.Request
{
	public class UserRequestTo
	{
		public int Id { get; set; }
		public string Login { get; set; }
		public string Password { get; set; }
		public string FirstName { get; set; }
		public string LastName { get; set; }
	}
}

File: 151001/Yankova/RV2/src/main/java/com/bsuir/nastassiayankova/beans/request/UserRequestTo.java
package com.bsuir.nastassiayankova.beans.request;

import jakarta.validation.constraints.Size;
import com.bsuir.nastassiayankova.beans.entity.ValidationMarker;

public record UserRequestTo(Long id,
                            @Size(min = 2, max = 64, groups = {ValidationMarker.OnCreate.class,
                                    ValidationMarker.OnUpdate.class}) String login,
                            @Size(min = 8, max = 128, groups = {ValidationMarker.OnCreate.class,
                                    ValidationMarker.OnUpdate.class}) String password,
                            @Size(min = 2, max = 64, groups = {ValidationMarker.OnCreate.class,
                                    ValidationMarker.OnUpdate.class}) String firstname,
                            @Size(min = 2, max = 64, groups = {ValidationMarker.OnCreate.class,
                                    ValidationMarker.OnUpdate.class}) String lastname) {
}
